### PR TITLE
ci(release): use new branch naming strategy

### PR DESCRIPTION
**Pull request by smoya**
Relates to https://github.com/asyncapi/spec/issues/734#issuecomment-1077455953

This PR changes the `.releaserc` file to reflect the changes on the branch naming strategy for releasing; using `next-spec` and `next-major-spec` branches.
This PR also updates the `RELEASE_PROCESS.md` accordingly. It also removes the step where the Release Coordinator should ask the Code Owners to create the release branch + the step where those should be added into `.releaserc` and/or `package.json` as they are not needed anymore.

### CODE REVIEW

**Review by dalelane — APPROVED** [review]
love how this made the process shorter :-)

**Review by derberg — DISMISSED** [review]

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -84,33 +84,29 @@ A [template for a new release issue](https://github.com/asyncapi/spec/blob/maste
...
 - [spec](https://github.com/asyncapi/spec) where contributor works with the specification file,
 - [spec-json-schemas](https://github.com/asyncapi/spec-json-schemas) where contributor pushes changes to JSON Schema of the spec,
 - [parser-js](https://github.com/asyncapi/parser-js) where contributor makes necessary changes in the JavaScript Parser.
+ 
+Depending on the type of release you are working on, you should use one branch or another:
 
-The [release coordinator](#%22release-coordinator%22) should decide what the branch name needs to be, and contact the [code owners](#code-owners) for each repository to ask them to create the branches for them.
-
+- `next-spec`: for releasing minor versions. For example, `2.4.0`.
+- `next-major-spec`: for releasing major versions. For example, `3.0.0`.
```
Comment (fmvilas) [inline RELEASE_PROCESS.md:98]:
```suggestion
- `next-spec`: for releasing minor versions. For example, `2.3.0` to `2.4.0`.
- `next-major-spec`: for releasing major versions. For example, `2.4.0` to `3.0.0`.
```

**Review by fmvilas — CHANGES_REQUESTED** [review]
Just some minor (no pun intended) suggestion

**Review by fmvilas — APPROVED** [review]

**Review by dalelane — APPROVED** [review]
LGTM

**Review by derberg — APPROVED** [review]

**Comment by fmvilas** [comment]
Forget about the last comment saying it was released in version 2.5.0-next-major-spec.1. I made a mistake and it created this version but it should actually be `3.0.0-next-major-spec.1`. There's a notice in the [release](https://github.com/asyncapi/spec/releases/tag/v2.5.0-next-major-spec.1). Apologies for the noise.

**Comment by derberg** [comment]
Recent comments about the release from the bot were added by mistake. More details in https://github.com/asyncapi/spec/issues/899

**Pull request merged**